src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,147 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INIT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INIT_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.key != null && current.key.equals(key)) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            current = table[hash];
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        size++;
+        if ((double) size / table.length > LOAD_FACTOR) {
+            resize();
+        }
+    }
+
+    private void putNullKey(V value) {
```

Comment:
method should be used to encapsulate duplication code and for reuse purposes.
`#putNullKey` method partially duplicates `#put` method logic, and should be merged into one.